### PR TITLE
syslog: Change g_syslog_default_sem to static function variable

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Impact
No, code refactor

## Testing
Pass CI

### CODE REVIEW

File: drivers/syslog/syslog_channel.c [hunk 1]
```diff
@@ -184,6 +180,7 @@ static ssize_t syslog_default_write(FAR struct syslog_channel_s *channel,
                                     FAR const char *buffer, size_t buflen)
 {
 #if defined(CONFIG_ARCH_LOWPUTC)
+  static sem_t g_syslog_default_sem = SEM_INITIALIZER(1);
```
Comment (pkarashchenko) [inline drivers/syslog/syslog_channel.c:183]:
```suggestion
  static sem_t default_sem = SEM_INITIALIZER(1);
```

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Please remove Gerrit change-id

File: drivers/syslog/syslog_channel.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/syslog/syslog_channel.c]: Done.

**Comment by xiaoxiang781216** [comment]
> Please remove Gerrit change-id

Done.

**Pull request merged**